exchanges/okx/okx.go

```diff
@@ -1990,161 +1802,195 @@ func (ok *Okx) GetBillsDetail(ctx context.Context, arg *BillsDetailQueryParamete
-func (ok *Okx) GetMaximumAvailableTradableAmount(ctx context.Context, instrumentID, currency, tradeMode string, reduceOnly bool, price float64) ([]MaximumTradableAmount, error) {
-	params := url.Values{}
-	if instrumentID != "" {
-		params.Set("instId", instrumentID)
-	} else {
-		return nil, errors.New("missing instrument id")
+func (ok *Okx) GetMaximumAvailableTradableAmount(ctx context.Context, ccy currency.Code, instrumentID, tradeMode, quickMarginType string, reduceOnly, upSpotOffset bool, price float64) ([]MaximumTradableAmount, error) {
+	if instrumentID == "" {
+		return nil, errMissingInstrumentID
 	}
-	if currency != "" {
-		params.Set("ccy", currency)
+	if tradeMode == "" {
+		// Supported tradeMode values are 'cross', 'isolated', 'cash', and 'spot_isolated'
+		return nil, fmt.Errorf("%w, possible values are 'cross', 'isolated', 'cash', and 'spot_isolated'", errInvalidTradeModeValue)
 	}
-	if tradeMode != TradeModeIsolated &&
-		tradeMode != TradeModeCross &&
-		tradeMode != TradeModeCash {
-		return nil, errInvalidTradeModeValue
+	params := url.Values{}
+	params.Set("instId", instrumentID)
+	params.Set("tdMode", tradeMode)
+	if !ccy.IsEmpty() {
+		params.Set("ccy", ccy.String())
 	}
 	if reduceOnly {
 		params.Set("reduceOnly", "true")
 	}
-	params.Set("tdMode", tradeMode)
-	params.Set("px", strconv.FormatFloat(price, 'f', 0, 64))
+	if price != 0 {
+		params.Set("px", strconv.FormatFloat(price, 'f', 0, 64))
+	}
+	if quickMarginType != "" {
+		params.Set("quickMgnType", quickMarginType)
+	}
+	if upSpotOffset {
+		params.Set("upSpotOffset", "true")
+	}
 	var resp []MaximumTradableAmount
-	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getMaximumAvailableTradableAmountEPL, http.MethodGet, common.EncodeURLValues(accountMaxAvailSize, params), nil, &resp, true)
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getMaximumAvailableTradableAmountEPL, http.MethodGet, common.EncodeURLValues("account/max-avail-size", params), nil, &resp, request.AuthenticatedRequest)
 }
 
-// IncreaseDecreaseMargin Increase or decrease the margin of the isolated position. Margin reduction may result in the change of the actual leverage.
+// IncreaseDecreaseMargin Increase or decrease the margin of the isolated position. Margin reduction may result in the change of the actual leverage
 func (ok *Okx) IncreaseDecreaseMargin(ctx context.Context, arg *IncreaseDecreaseMarginInput) (*IncreaseDecreaseMargin, error) {
+	if *arg == (IncreaseDecreaseMarginInput{}) {
+		return nil, common.ErrEmptyParams
+	}
 	if arg.InstrumentID == "" {
 		return nil, errMissingInstrumentID
 	}
-	if arg.PositionSide != positionSideLong &&
-		arg.PositionSide != positionSideShort &&
-		arg.PositionSide != positionSideNet {
-		return nil, errors.New("missing valid position side")
+	if !slices.Contains([]string{positionSideLong, positionSideShort, positionSideNet}, arg.PositionSide) {
+		return nil, fmt.Errorf("%w, position side is required", order.ErrSideIsInvalid)
```

Comment:
Misleading message, because we possible have a side, it just isn't what you want.
Need to confirm the side that's wrong.
```suggestion
		return nil, fmt.Errorf("%w: `%s`", order.ErrSideIsInvalid, arg.PositionSide)
```